docs/src/main/asciidoc/_admin-guide/subsystem-configuration/MicroProfile_Config_SmallRye.adoc

```diff
@@ -145,6 +145,54 @@ This results in the XML configuration:
 
 All properties from the `ConfigSource`s provided by this `ConfigSourceProvider` will be available to any WildFly deployment.
 
+=== ConfigSourceProvider from Root directory
+You can create a `ConfigSourceProvider` for a root directory by creating a `config-source-root` resource pointing to a
+root directory. Top level directories within this root directory each become an individual `ConfigSource` reading from a
+directory similar to what we saw earlier in <<configsource-from-directory, ConfigSource from Directory>>.
+
+This is especially useful when running on OpenShift where `ConfigMap` and `ServiceBinding` instances get mapped under
+a common known location. For example if there are two `ConfigMap` instances (for this example, we will call them
+`map-a`, and `map-b`) used by your application pod, they will
+each get mapped under `/etc/config`. So you would have `/etc/config/map-a` and `/etc/config/map-b` directories. Each of
+these directories will have files where the file name is the name of the property and the file content is the value of
+the property, like we saw earlier.
```

Comment:
should we mentioned that any directories below the top-level directories are ignored?

iirc some implementation of mp config allowed to have deep-nested file representation, eg
```
$ cat a/b/c/foo
bar
```

would be equivalent to the prop `a.b.c.foo=bar`